src/javaRosa.js

```diff
@@ -1257,64 +1257,49 @@ define([
     };
 
     var generateItextXLS = function (vellum, Itext) {
-        // todo: fix abstraction barrier
-        vellum.beforeSerialize();
-
-        // TODO move TSV generation logic into tsv module
         function getItemFormValues(item, languages, form) {
-
-            var ret = [];
-
-            for(var i = 0; i < languages.length; i++) {
-                var language = languages[i];
-                var value = item.hasForm(form) ? (item.getForm(form).getValueOrDefault(language) || "") : "";
-
-                if (specialChars.test(value)) {
-                    // quote field
-                    value = '"' + value.replace(/"/g, '""') + '"';
-                }
-                ret.push(value);
+            var ret = [], i, language;
+            for(i = 0; i < languages.length; i++) {
+                language = languages[i];
+                ret.push(item.hasForm(form) ? item.get(language, form) : "");
             }
-            return ret.join("\t");
+            return ret;
         }
 
-        function makeRow (languages, item, forms) {
-
-            var questions = getItemFormValues(item, languages, forms[0]);
-            var audios = getItemFormValues(item, languages, forms[1]);
-            var images = getItemFormValues(item, languages, forms[2]);
-            var videos = getItemFormValues(item, languages, forms[3]);
-
-            var row = [item.id, questions, audios, images, videos];
-            return row.join("\t");
+        function makeRow(languages, item, forms) {
+            var row = [item.id],
+                extend = Array.prototype.push.apply.bind(Array.prototype.push);
+            for (var i = 0; i < forms.length; i++) {
+                extend(row, getItemFormValues(item, languages, forms[i]));
+            }
+            return row;
         }
 
         function makeHeadings(languages, exportCols) {
-            var header_row = ["label"];
-            for(i = 0; i < exportCols.length; i++) {
-                for(j=0; j < languages.length; j++) {
+            var header_row = ["label"], i, j;
+            for (i = 0; i < exportCols.length; i++) {
+                for (j = 0; j < languages.length; j++) {
                     header_row.push(exportCols[i] + '-' + languages[j]);
                 }
             }
-            return header_row.join("\t");
+            return header_row;
         }
 
-        var specialChars = /[\r\n\u2028\u2029"]/g;
```

Comment:
This (specifically the `g` flag) was the problem for bulk translations. Global RegExp objects are stateful, which caused some cells to not be escaped even though they needed it!

All other modifications to this file are cleanup and refactoring to use the tsv module for TSV encoding.